marconi/test/Integration.hs

```diff
@@ -296,7 +296,13 @@ testIndex = H.integration . HE.runFinallies . workspace "chairman" $ \tempAbsBas
   because the indexer runs in a separate thread and there is no way
   of awaiting the data to be flushed into the database. -}
 
-  (ScriptTx.TxCbor tx, indexedScriptHashes) :| _ <- liftIO $ IO.readChan indexedTxs
+  indexedWithScriptHashes <- liftIO $ IO.readChan indexedTxs
+
+  -- We have to filter out the txs the empty scripts hashes because
+  -- sometimes the RollForward event contains a block with the first transaction 'tx1'
+  -- which has no scripts. The test fails because of that in 'headM indexedScriptHashes'.
+  -- For more details see https://github.com/input-output-hk/plutus-apps/issues/775
+  let (ScriptTx.TxCbor tx, indexedScriptHashes) = head $ (NE.filter (\(_, hashes) -> hashes /= [])) indexedWithScriptHashes
```

Comment:
very Nice catch.
Shouldn't we fix the [headM](https://github.com/input-output-hk/plutus-apps/blob/fix-marconi-test/marconi/test/Integration.hs#L370) function and throwError when the list is empty?

Reply:
We need your fix, so I have no question regarding that.
My main question is if the `headM []` behaves correctly?
On my machine, Linux, without your fix, the tests just freezes. It doesn't advance. 
If we change headM[] to throwError that would fail the test instead of the current behavior, no?